src/wp-includes/class-wp-term-query.php

```diff
@@ -836,54 +841,24 @@ public function get_terms() {
 			$terms = $_terms;
 		}
 
-		$_terms = array();
-		if ( 'id=>parent' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[ $term->term_id ] = $term->parent;
-			}
-		} elseif ( 'ids' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = (int) $term->term_id;
-			}
-		} elseif ( 'tt_ids' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = (int) $term->term_taxonomy_id;
-			}
-		} elseif ( 'names' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = $term->name;
-			}
-		} elseif ( 'slugs' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[] = $term->slug;
-			}
-		} elseif ( 'id=>name' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[ $term->term_id ] = $term->name;
-			}
-		} elseif ( 'id=>slug' === $_fields ) {
-			foreach ( $terms as $term ) {
-				$_terms[ $term->term_id ] = $term->slug;
-			}
-		}
-
-		if ( ! empty( $_terms ) ) {
-			$terms = $_terms;
-		}
-
 		// Hierarchical queries are not limited, so 'offset' and 'number' must be handled now.
 		if ( $hierarchical && $number && is_array( $terms ) ) {
 			if ( $offset >= count( $terms ) ) {
-				$terms = array();
+				$terms         = array();
+				$terms_objects = array();
 			} else {
-				$terms = array_slice( $terms, $offset, $number, true );
+				$terms         = array_slice( $terms, $offset, $number, true );
+				$terms_objects = array_slice( $terms_objects, $offset, $number, true );
 			}
 		}
 
-		wp_cache_add( $cache_key, $terms, 'terms' );
-
-		if ( 'all' === $_fields || 'all_with_object_id' === $_fields ) {
+		if ( 'all_with_object_id' === $_fields ) {
+			wp_cache_add( $cache_key, $terms, 'terms' );
```

Comment:
Why are the entire terms stored in the `...with_object_id` use case rather than just the IDs?

Reply:
Making that work was pretty hard, but it is possible. We would need to store the term id and object id. When create loop around WP_term objects and adding the object id to those objects.

Reply:
Fixed this now.